### PR TITLE
ENH: Supporting custom kernel in local linear kernel regression

### PR DESCRIPTION
**Pull request by vsriram11**
Currently, KernelRegression supports only the Gaussian kernel. By adding the kernel as a parameter to the class, we can support custom kernels for the local linear mode of kernel regression.

This is useful for performing other nonparametric regressions. For example, this would now support the LOESS regression, which can be achieved by specifying local linear mode, and a tricube kernel. 

This PR also adds the tricube kernel to statsmodels library of kernels.

### CODE REVIEW

File: statsmodels/nonparametric/kernel_regression.py [hunk 1]
```diff
@@ -83,10 +91,14 @@ class KernelReg(GenericKDE):
         The bandwidth parameters.
     """
     def __init__(self, endog, exog, var_type, reg_type='ll', bw='cv_ls',
-                 defaults=None):
+                 ckertype='gaussian', okertype='wangryzin',
+                 ukertype='aitchisonaitken', defaults=None):
         self.var_type = var_type
         self.data_type = var_type
         self.reg_type = reg_type
+        self.ckertype = ckertype
```
Comment (bashtage) [inline statsmodels/nonparametric/kernel_regression.py]: Can you add a check that these are value kernel names, and raise if not with a meaningful error.

Comment (vsriram11) [reply statsmodels/nonparametric/kernel_regression.py]: Sure done, and added relevant unit test

Comment (vsriram11) [reply statsmodels/nonparametric/kernel_regression.py]: @bashtage Any idea why the build checks below fail now though? Looks like it's failing when collecting tests, but it works fine locally..

Comment (vsriram11) [reply statsmodels/nonparametric/kernel_regression.py]: Never mind, figured it out. Looks like a documented issue with pytest fixtures, that I managed to work around

**Comment by bashtage** [comment]
Looks great, thanks.

**Pull request merged**